external/js/kyber/lib/group/nist/point.js

```diff
@@ -0,0 +1,357 @@
+  this.ref.point = this.ref.curve.curve
+    .point(point.x, point.y)
+    .add(p2.ref.point);
+  return this;
+};
+
+/**
+ * Subtract two points
+ * Modifies the receiver
+ *
+ * @param {module:group/nist~Point} p1 Point object
+ * @param {module:group/nist~Point} p2 Point object
+ * @return {module:group/nist~Point} p1 - p2
+ */
+Point.prototype.sub = function(p1, p2) {
+  const point = p1.ref.point;
+  this.ref.point = this.ref.curve.curve
+    .point(point.x, point.y)
+    .add(p2.ref.point.neg());
+  return this;
+};
+
+/**
+ * Finds the negative of a point p
+ * Modifies the receiver
+ *
+ * @param {module:group/nist~Point} p Point to negate
+ * @return {module:group/nist~Point} -p
+ */
+Point.prototype.neg = function(p) {
+  this.ref.point = p.ref.point.neg();
+  return this;
+};
+
+/**
+ * Multiply point p by scalar s.
+ * If p is not passed then multiplies the base point of the curve with
+ * scalar s
+ * Modifies the receiver
+ *
+ * @param {module:group/nist~Scalar} s Scalar
+ * @param {module:group/nist~Point} [p=null] Point
+ * @return {module:group/nist~Point}
+ */
+Point.prototype.mul = function(s, p) {
+  if (s.constructor !== Scalar) {
+    throw new TypeError();
+  }
+  p = p || null;
+  const arr = s.ref.arr.fromRed();
+  this.ref.point =
+    p !== null ? p.ref.point.mul(arr) : this.ref.curve.curve.g.mul(arr);
+  return this;
+};
+
+/**
+ * Selects a random point
+ *
+ * @param {function} [callback] - to generate a random byte array of given length
+ * @return {module:group/nist~Point}
+ */
+Point.prototype.pick = function(callback) {
+  return this.embed(new Uint8Array(), callback);
+};
+
+Point.prototype.marshalSize = function() {
+  // uncompressed ANSI X9.62 representation
+  return this.ref.curve.pointLen();
+};
+
+/**
+ * converts a point into the form specified in section 4.3.6 of ANSI X9.62.
+ *
+ * @return {Uint8Array} byte representation
+ */
+Point.prototype.marshalBinary = function() {
+  const byteLen = this.ref.curve._coordLen();
+  let buf = new Uint8Array(this.ref.curve.pointLen());
+  buf[0] = 4; // uncompressed point
+
+  let xBytes = this.ref.point.x.fromRed().toArray("be");
+  buf.set(xBytes, 1 + byteLen - xBytes.length);
+  let yBytes = this.ref.point.y.fromRed().toArray("be");
+  buf.set(yBytes, 1 + 2 * byteLen - yBytes.length);
+
+  return buf;
+};
+
+/**
+ * Convert a Uint8Array back to a curve point
+ * @param {Uint8Array} bytes
+ *
+ * @throws {TypeError} when bytes is not Uint8Array
+ * @throws {Error} when bytes does not correspond to a valid point
+ */
+Point.prototype.unmarshalBinary = function(bytes) {
+  if (bytes.constructor !== Uint8Array) {
+    throw new TypeError();
+  }
+  const byteLen = this.ref.curve._coordLen();
+  if (bytes.length != 1 + 2 * byteLen) {
+    throw new Error();
+  }
+  if (bytes[0] != 4) {
+    throw new Error();
+  }
+  let x = new BN(bytes.slice(1, 1 + byteLen), 16);
+  let y = new BN(bytes.slice(1 + byteLen), 16);
+  let zero = new BN(0);
```

Comment:
Is there a way in javascript to put a constant like this in a place where it is initialized one single time?